aten/src/ATen/native/cuda/MiscUtils.h

```diff
@@ -0,0 +1,64 @@
+#include "ATen/ATen.h"
+#include "THC.h"  // for USE_MAGMA
+
+#ifdef USE_MAGMA
+#include <magma.h>
+#include <magma_types.h>
+#endif
+
+namespace at {
+namespace native {
+
+#ifdef USE_MAGMA
+
+// RAII for a MAGMA Queue
+struct MAGMAQueue {
```

Comment:
@zou3519 I've added the RAII structure. Please let me know if this is what you had in mind.